contracts/contracts/vault/VaultCore.sol

```diff
@@ -224,7 +224,7 @@ contract VaultCore is VaultInitializer {
         }
 
         // Check that the OTokens are backed by enough assets
-        if (maxSupplyDiff > 0) {
+        if (maxSupplyDiff > 0 && totalUnits > 0) {
```

Comment:
🟥 This skips the solvency check if an attacker has managed to steal all the money. That's the wrong time to skip the solvency check.

Now in practice, we of course don't think that an attacker could steal all the money, but still, this function should catch it if they did.

Reply:
It also skips the solvency check if `totalUnits` is zero because the vault value is less than the outstanding withdrawal requests. This can happen when there is a mass slashing and most users have requested a withdrawal.

I need to consider how to handle this.

Reply:
I've changed `_postRedeem` in `VaultCore` to

```Solidity
// Check that the OTokens are backed by enough assets
if (maxSupplyDiff > 0) {
    // If there are more outstanding withdrawal requests than assets in the vault and strategies
    // then the available assets will be negative and totalUnits will be rounded up to zero.
    // As we don't know the exact shortfall amount, we will reject all redeem and withdrawals
    require(totalUnits > 0, "Too many outstanding requests");


```